src/v/config/rest_authn_endpoint.cc

```diff
@@ -0,0 +1,110 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "config/rest_authn_endpoint.h"
+
+#include "kafka/client/exceptions.h"
+#include "model/metadata.h"
+#include "utils/string_switch.h"
+
+namespace config {
+
+constexpr inline std::string_view to_string_view(rest_authn_type m) {
+    switch (m) {
+    case rest_authn_type::none:
+        return "none";
+    case rest_authn_type::http_basic:
+        return "http_basic";
+    default:
+        return "invalid";
```

Comment:
I usually prefer not to set a default case here; then the compiler will warn for missing cases.

The `return "invalid";` can be the last line of the function, outside of the switch.